exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+	params.Set("recvWindow", recvWindowSize5000String)
+	var rateLimit request.EndpointLimit
+	if symbol != "" {
+		rateLimit = spotOpenOrdersSpecificRate
+	} else {
+		rateLimit = spotOpenOrdersAllRate
+	}
+	return response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary, http.MethodGet,
+		openOrders, params,
+		rateLimit, &response)
+}
+
+// CancelExistingOrder to cancel an active trade order.
+func (bi *Binanceus) CancelExistingOrder(ctx context.Context, arg *CancelOrderRequestParams) (*Order, error) {
+	params := url.Values{}
+	var response Order
+	params.Set("timestamp", strconv.FormatInt(time.Now().UnixMilli(), 10))
+	symbolValue, err := bi.FormatSymbol(arg.Symbol, asset.Spot)
+	if err != nil || symbolValue == "" {
+		return nil, errMissingCurrencySymbol
+	}
+	params.Set("symbol", symbolValue)
+	if arg.OrderID == "" && arg.ClientSuppliedOrderID == "" {
+		return nil, errEitherOrderIDOrClientOrderIDIsRequired
+	}
+	if arg.ClientSuppliedOrderID != "" {
+		params.Set("origClientOrderId", arg.ClientSuppliedOrderID)
+	} else {
+		params.Set("orderId", arg.OrderID)
+	}
+	params.Set("recvWindow", recvWindowSize5000String)
+	return &response, bi.SendAuthHTTPRequest(ctx,
+		exchange.RestSpotSupplementary,
+		http.MethodDelete, orderRequest,
+		params, spotDefaultRate, &response)
+}
+
+// CancelOpenOrdersForSymbol request to cancel an open orders.
+func (bi *Binanceus) CancelOpenOrdersForSymbol(ctx context.Context, symbol string) ([]Order, error) {
+	params := url.Values{}
+	if symbol == "" || len(symbol) < 4 {
+		return nil, errMissingCurrencySymbol
+	}
+	params.Set("symbol", symbol)
+	params.Set("timestamp", strconv.Itoa(int(time.Now().UnixMilli())))
+	params.Set("recvWindow", "5000")
+	var response []Order
+	return response, bi.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary,
+		http.MethodDelete, openOrders,
+		params, spotDefaultRate, response)
+}
+
+// GetTrades to get trade data for a specific account and token symbol.
+func (bi *Binanceus) GetTrades(ctx context.Context, arg *GetTradesParams) ([]*Trade, error) {
+	var resp []*Trade
+	params := url.Values{}
+	if arg.Symbol == "" || len(arg.Symbol) <= 2 {
+		return nil, errIncompleteArguments
+	}
+	params.Set("symbol", arg.Symbol)
+	params.Set("timestamp", strconv.Itoa(int(time.Now().UnixMilli())))
+	if arg.RecvWindow > 3000 {
+		params.Set("recvWindow", strconv.Itoa(int(arg.RecvWindow)))
+	}
+	if arg.StartTime != nil {
+		params.Set("startTime", strconv.Itoa(int(arg.StartTime.UnixMilli())))
+	}
+	if arg.EndTime != nil {
+		params.Set("endTime", strconv.Itoa(int(arg.EndTime.UnixMilli())))
+	}
+	if arg.FromID > 0 {
+		params.Set("fromId", strconv.Itoa(int(arg.FromID)))
+	}
+	if arg.Limit > 0 && arg.Limit < 1000 {
+		params.Set("limit", fmt.Sprint(arg.Limit))
+	} else if arg.Limit > 1000 {
+		params.Set("limit", strconv.Itoa(1000))
+	}
+	return resp, bi.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, myTrades, params, spotDefaultRate, &resp)
```

Comment:
GetTrades endpoint has a ratelimit weight of 10, not 1 and is for the orders rate limit. See https://docs.binance.us/#get-trades